### PR TITLE
fixing idempotence on hab_sup_systemd.

### PR DESCRIPTION
**Pull request by sam1el**
Signed-off-by: Jeff Brimager <jbrimager@chef.io>

When running this in effortless. the hab_sup resources tries to reinstall every time. We have only witnessed this on systemd so far.  

### Description
Adds guard to hab_sup_systemd
Bumping version to 1.5.50

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

File: libraries/resource_hab_sup_systemd.rb [hunk 1]
```diff
@@ -63,4 +63,4 @@ class HabSupSystemd < HabSup
       end
     end
   end
-end
+end unless system('hab sup status')
```
Comment (tas50) [inline libraries/resource_hab_sup_systemd.rb]: You def don't want to shell out with system in a cookbook

Comment (sam1el) [reply libraries/resource_hab_sup_systemd.rb]: Understood. do you have thoughts on a better guard? we will likely have to do this on every service resource

Comment (tas50) [reply libraries/resource_hab_sup_systemd.rb]: can you explain exactly what the problem you're trying to solve for is?

Comment (sam1el) [reply libraries/resource_hab_sup_systemd.rb]: When running this cookbook in effortless. the supervisor resources are failing every converge after the initial. It's looking like the version pinning is the what's causing the issue.

Comment (sam1el) [reply libraries/resource_hab_sup_systemd.rb]: confirmed.. version pinnings were all mismatched. causes the sup and service resources to get a little crazy. I'd like to find a way to set them to always latest and let the customer manger their versions

**Pull request merged**